core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -282,6 +282,12 @@ private[coordinator] class GroupMetadata(val groupId: String, initialState: Grou
     pendingOffsetCommits ++= offsets
   }
 
+  def removeOffsets(topicPartition: TopicPartition) = {
+    val removedOffsets = offsets.filter( _._1 == new TopicPartition(topicPartition.topic, topicPartition.partition))
```

Comment:
Not sure I get why all this logic is needed. Couldn't we just do `offsets.remove(topicPartition)`? That will return an `Option` with the previously stored offset (if there was one). 

One bit of trickiness is that we also need to clean up the `pendingOffsetCommits` collection. But if we have an offset commit pending, then once it completes, we will (incorrectly) add the offset to the cache. Not sure the best way to deal with this. Perhaps one way is to keep a set of the deleted partitions which have pending offset commits. Then when the pending offset commit completes, we can check that collection prior to inserting it into the cache.